pallets/xyk/src/lib.rs

```diff
@@ -346,7 +345,7 @@ pub mod pallet {
 				T::MinPoolLiquidity::get(),
 			)?;
 
-			let _ = T::AMMHandler::on_create_pool(asset_pair.asset_in, asset_pair.asset_out)?;
+			T::AMMHandler::on_create_pool(asset_pair.asset_in, asset_pair.asset_out)?;
```

Comment:
i am not sure about this.

we should not really fail to create a pool when something outside xyk pallet fails ? or ?

Reply:
The only change is that we removed `let _`. Clippy failed on that. Is there something wrong with it?

Reply:
now, with this change,  extrinsic fails. before it ignored the error.

Reply:
Ahh. I missed the question mark. It shouldn't be there

Reply:
Do we want to include this change in this PR?

Reply:
i would create an issue to track this . create pool should not fail if the callback fails imo.